editor/src/components/inspector/controls/slider-control.tsx

```diff
@@ -20,7 +20,101 @@ export type SliderControlProps = DEPRECATEDControlProps<number> & {
   DEPRECATED_controlOptions: DEPRECATEDSliderControlOptions
 }
 
-export class SliderControl extends React.Component<SliderControlProps> {
+export const SliderControl: React.FunctionComponent<SliderControlProps> = (props) => {
+  const [isSliding, setIsSliding] = React.useState(false)
+
+  const handleBeforeChange = React.useCallback(() => {
+    setIsSliding(true)
+  }, [])
+
+  const onChangeFn = props.onForcedSubmitValue ?? props.onSubmitValue
+
+  const handleOnAfterChange = React.useCallback(
+    (n: number) => {
+      onChangeFn(n)
+      setIsSliding(false)
+    },
+    [onChangeFn],
+  )
+
+  const controlOptions = {
```

Comment:
I believe these variables will need wrapping in `useMemo` invocations, otherwise they'll be regenerated every time.